### ISSUE TITLE
Docs: [class-methods-use-this]  ignoreClassesThatImplementAnInterface public-fields option's description and playground link is not clear

### ISSUE THREAD

**Issue by developer-bandi** [typescript-eslint/typescript-eslint#8729]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

In [class-methods-use-this's ignoreClassesThatImplementAnInterface options's public-fields docs](https://typescript-eslint.io/rules/class-methods-use-this/#public-fields), I found two errors.

1. description is not clear
the description is `Example of incorrect code when ignoreClassesThatImplementAnInterface is set to 'public-fields':` but, below example code is incorrect and correct code. I think only correct code is enough to explain the option and above ignoreClassesThatImplementAnInterface's true example should fix.

result is in below

ignoreClassesThatImplementAnInterface's value is true
```javascript
class X implements Y {
  method() {};
  property = () => { };

  private privateMethod() { };
  private privateProperty = () => { };

  protected protectedMethod() { } ;
  protected protedtedProperty = () => { };
}
```

ignoreClassesThatImplementAnInterface's value is public-fields
```javascript
class X implements Y {
  method() {}
  property = () => {};
}
```

2. playground link
correct and incorrect error's [playground link](https://typescript-eslint.io/play/#eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQMbwENlkBaAW0QBcALAewBNTZlESaBLVDSWgM18gAacNkgABSgE8ADilzR20yiRTx2AO0oB6fEVIUaDJizbVO6KImjRa0SCIC%2BIB0A&code=MYGwhgzhAEAa0EsC2AHEBTJ6B2AXGAmtAN4BQ00WuAFgPYAmAFAJQkC%2B50KATrSut1wBPaAF5oLMQD52AblKceCAG5hc6LtxVr0AWXQ0Gk4hwpLV6zdvUAFXv0Ejxk0TJPzFvdcHX0rFvQM6JlYTT1pvX38dOz4BYTEJVlc5UjYgA&fileType=.ts) is not include ignoreClassesThatImplementAnInterface option that is public-fields

### Affected URL(s)

https://typescript-eslint.io/rules/class-methods-use-this/#public-fields

**Comment by Josh-Cena** [comment]
I think they mean:

- The text says "Example of incorrect code" but we show tabs with both incorrect and correct code, so that's inconsistent.
- "Incorrect code" is not really necessary, since they are incorrect under the default options anyway. The option only disables certain errors, so we just need to show the correct versions (which I disagree—it's better to compare side by side which cases continue to error and which cases don't).

`ignoreClassesThatImplementAnInterface: true` reports a subset of cases than `ignoreClassesThatImplementAnInterface: 'public-fields'` than `ignoreClassesThatImplementAnInterface: false`, so to make the docs least confusing, I think the tabs should be organized like this:

- Code that remains errors under `ignoreClassesThatImplementAnInterface: true`
- Code that is correct under `ignoreClassesThatImplementAnInterface: true` but errors under `ignoreClassesThatImplementAnInterface: 'public-fields'`
- Code that is correct under `ignoreClassesThatImplementAnInterface: 'public-fields'` but errors under `ignoreClassesThatImplementAnInterface: false`
